src/bindings/swig/python/examples/example_gopts.py

```diff
@@ -0,0 +1,160 @@
+		  kdb.Key(spec_base_key + "/setter/verbose",
+			  kdb.KEY_META, "description", "print additional information about where the value will be stored",
+			  kdb.KEY_META, "opt", "v",
+			  kdb.KEY_META, "opt/long", "verbose",
+			  kdb.KEY_META, "opt/arg", "none"),
+		  kdb.Key(spec_base_key + "/setter/keyname",
+			  kdb.KEY_META, "description", "name of the key to write",
+			  kdb.KEY_META, "args", "indexed",
+			  kdb.KEY_META, "args/index", "0"),
+		  kdb.Key(spec_base_key + "/setter/value",
+			  kdb.KEY_META, "description", "value to be written",
+			  kdb.KEY_META, "args", "indexed", kdb.KEY_META, "args/index", "1"),
+		  kdb.Key(spec_base_key + "/dynamic/#",
+			  kdb.KEY_META, "description", "dynamically call a user-supplied command",
+			  kdb.KEY_META, "args", "remaining")
+		  )
+
+
+def setupSpec():
+	with kdb.KDB() as db:
+		ks = kdb.KeySet(10)
+		db.get(ks, spec_base_key)
+		if len(ks.cut(kdb.Key(spec_base_key))) > 0:
+			print("ERROR: Couldn't setup spec, keys exist!", file=sys.stderr)
+			exit(1)
+		
+		ks.extend(spec)
+		db.set(ks, spec_base_key)
+
+def removeSpec():
+	with kdb.KDB() as db:
+		ks = kdb.KeySet(10)
+		db.get(ks, spec_base_key)
+		ks.cut(kdb.Key(spec_base_key))
+		db.set(ks, spec_base_key)
+
+def array_index(index):
+	return "_" * (len(str(index)) - 1) + str(index)
+
+def get_python_interpreter_arguments():
+	import ctypes
+	argc = ctypes.c_int()
+	argv = ctypes.POINTER(ctypes.c_wchar_p if sys.version_info >= (3, ) else ctypes.c_char_p)()
+	ctypes.pythonapi.Py_GetArgcArgv(ctypes.byref(argc), ctypes.byref(argv))
+
+	# Ctypes are weird. They can't be used in list comprehensions, you can't use `in` with them, and you can't
+	# use a for-each loop on them. We have to do an old-school for-i loop.
+	arguments = list()
+	for i in range(argc.value):
+		arguments.append(argv[i])
+
+	return arguments
+
+# we need to calculate the number of arguments given to the python interpreter itself
+# this number is given to 'gopts' as an offset, so it only looks at the keys actually relvant to this script
+argv_offset = len(get_python_interpreter_arguments()) - len(sys.argv)
+
+setupSpec()
+with kdb.KDB() as db:
+	error_key = kdb.Key(spec_base_key)
+	rc = db.ensure(kdb.KeySet(2,
+			kdb.Key("system/elektra/ensure/plugins/global/gopts", kdb.KEY_VALUE, "remount"), 
+			kdb.Key("system/elektra/ensure/plugins/global/gopts/config/offset", kdb.KEY_VALUE, str(argv_offset))
+			), error_key)
+	if rc == 1:
+		print("ERROR: Contract could not be ensured!\n", error_key.getMeta("error/reason").value, file=sys.stderr)
+		removeSpec()
+		exit(1)
+	elif rc == -1:
+		print("ERROR: Contract malformed/NULL pointer!\n", error_key.getMeta("error/reason").value, file=sys.stderr)
+		removeSpec()
+		exit(1)
+	
+	ks = kdb.KeySet(0)
+	db.get(ks, spec_base_key)
+	
+	if "proc/elektra/gopts/help" in ks and ks["proc/elektra/gopts/help"].value == "1":
+		# TODO: call elektraGetOptsHelpMessage (not exposed to python right now) and print result
```

Comment:
Did you consider to add the help text directly as value of "proc/elektra/gopts/help"?

Reply:
I used `proc/elektra/gopts/help/message` to maintain backwards compatibility.

Reply:
Very nice :fireworks: